test/stdlib/CodableTests.swift

```diff
@@ -257,9 +257,9 @@ class TestCodable : TestCodableSuper {
     ]
 
     func test_CharacterSet_JSON() {
-        for (testLine, characterSet) in characterSetValues {
-            expectRoundTripEqualityThroughJSON(for: characterSet, lineNumber: testLine)
-        }
+        // for (testLine, characterSet) in characterSetValues {
+        //     expectRoundTripEqualityThroughJSON(for: characterSet, lineNumber: testLine)
+        // }
```

Comment:
We probably shouldn't leave tests commented out.

Reply:
Here's a sample, in case it's helpful, but this is just for context. It's not related to the PR; just something I had to comment out to run tests locally using the method described above. I can file a bug if needed.

[Sample of a.out.txt](https://github.com/user-attachments/files/21583863/Sample.of.a.out.txt)

Reply:
update: I let the test run for 1h15min and it was still pegged at 100% CPU.

Reply:
I thought about bisecting it, but running git-bisect for this seemed more than a little daunting. Would I have to build the whole compiler each time? I wonder if there are any tricks folks have for tracking down these kinds of issues (besides familiarity with the code, that is).

Reply:
Try removing `--debug-swift-stdlib`? I'm told that builds the stdlib with `-Onone`, so none of the optimizations that we need in strings is happening there.

Reply:
That makes sense! It's amazing that it would have that much of an impact on run times. I'll give it a go.

🤔 I can't guarantee that this is the exact invocation I was using when I started the branch, but I'm 99% sure I used it to run these tests a few months ago, and that test didn't take ages to run.

Reply:
Without `--debug-swift-stdlib`, it builds, but `test_CharacterSet_JSON()` is still hanging or at least taking forever. Another sample in case it's useful:

[Sample of a.out.txt](https://github.com/user-attachments/files/21589942/Sample.of.a.out.txt)

Reply:
…but I ran with `--debug-swift-stdlib` enabled, and the tests finished in 23558.30s, i.e. 6.5 hours 😬. Worth opening a ticket?

Reply:
-Onone is known to often result in rather slow code. A ticket could potentially lead to an investigation of whether it could do better here. Alternatively, the test could be changed to either require an optimized stdlib or avoid / escaping so it hits the faster path…